### PR TITLE
Add SAMI parser

### PR DESCRIPTION
**Pull request by thiagopnts**
This adds a basic tag parser using `golang.org/x/net/html`'s tokenizer. The main goal here is parsing all the tags naively without having to infer/enforce any HTML5 spec restrictions/rules(which is what `html.Parse` will do if we feed it a SAMI file). Since we're still using `html.Node`s to represent the SAMI tree, we'll still be able to use something like `github.com/PuerkitoBio/goquery` to query and edit the tree in the SAMI reader/writer.

TLDR: Most of the code here is based on `golang.org/x/net/html` html parser, just removing all HTML5 semantic inference/enforcement. Example of what happens if we parse SAMI with the original html parser: https://go.dev/play/p/GTnOVGkcYsx

### CODE REVIEW

File: sami/parser.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	}
+	return nil
+}
+
+// nodeStack is a stack of nodes.
+type nodeStack []*html.Node
+
+// pop pops the stack. It will panic if s is empty.
+func (s *nodeStack) pop() *html.Node {
+	i := len(*s)
```
Comment (nakkamarra) [inline sami/parser.go:137]: What if len == 0

Comment (thiagopnts) [reply sami/parser.go:137]: fixed this, the panic on empty stack pop was intended as its in the stringdoc(this came from the html pkg unchanged), but I didn't know why. Now I get why its like this, the html parser will always infer closing tags, so the pop on empty panics because its a state that the parser should never be, which is not our case

**Review by nakkamarra — APPROVED** [review]
